ores/scoring_systems/scoring_system.py

```diff
@@ -240,6 +255,35 @@ def _lookup_cached_score(self, request, rev_id, model_name):
             self.metrics_collector.score_cache_miss(request, model_name)
             raise
 
+    def _connect_to_lock_manager(self):
+        try:
+            self.lock_manager.connect()
+        # Lock manager is down, let's do nothing
+        except ConnectionRefusedError:
+            logger.warning('Can not connect to lock manager')
+            return False
+        return True
+
+    def _lock_ip(self, ip):
```

Comment:
IMO this should be generalized, so we can lock on things other than just IP.  For example, we might have a separate pool with different configuration for requests that include features.